### PR TITLE
Fix #4459 by improving `par:` error message

### LINKED ISSUE #4459: "The solution is not ground" is a horrible way of indicating that a proof has unresolved existential variables
Note: the issue was created automatically with bugzilla2github tool

Original bug ID: BZ#4459
From: @JasonGross
Reported version: 8.6
CC:   @gares

### PR DESCRIPTION
**Pull request by maximedenes**

### CODE REVIEW

File: stm/stm.ml [hunk 1]
```diff
@@ -1950,10 +1949,15 @@ end = struct (* {{{ *)
...
-            if Evarutil.is_ground_term sigma t then
+            let evars = Evarutil.undefined_evars_of_term sigma t in
+            if Evar.Set.is_empty evars then
               let t = EConstr.Unsafe.to_constr t in
               RespBuiltSubProof (t, Evd.evar_universe_context sigma)
-            else CErrors.user_err ~hdr:"STM" Pp.(str"The solution is not ground")
+            else
+              CErrors.user_err ~hdr:"STM"
+                Pp.(str"The par: selector requires a tactic that makes no progress or fully" ++
+                    str" solves all goals and leaves no unresolved existentials. The following" ++
```
Comment (SkySkimmer) [inline stm/stm.ml:1959]: I wouldn't say that the tactic solves all goals, it only solves the 1 goal it is given and leaves the other to the parallel runs.

Comment (gares) [reply stm/stm.ml:1959]:
```suggestion
                    str" solves the goal and leaves no unresolved existential variables. The following" ++
```

**Review by gares — APPROVED** [review]
OK, consider merging the suggestion

**Comment by Zimmi48** [comment]
Feel free to change the milestone to 8.12.0: error message improvements are OK to backport.

**Pull request merged**